src/main/java/tool/designpatterns/verifiers/multiclassverifiers/decorator/DecoratorVerifier.java

```diff
@@ -0,0 +1,224 @@
+        if (hasAComponent.get()) {
+            result = Feedback.getSuccessfulFeedback();
+        } else {
+            result = Feedback.getNoChildFeedback(
+                "There was no field with type '" + type.getNameAsString() + "'",
+                new FeedbackTrace(toTest));
+        }
+        return result;
+    }
+
+    /**
+     * Checks if all constructors in a given class initialize the Component field in the class.
+     *
+     * @param toTest        The class to check.
+     * @param interfaceName
+     *
+     * @return A {@link Feedback} object containing the result
+     */
+    private Feedback fieldInitializedInConstr(
+        ClassOrInterfaceDeclaration toTest, ClassOrInterfaceDeclaration interfaceName) {
+        Feedback result;
+        AtomicBoolean isInitialized = new AtomicBoolean(true);
+        List<FieldDeclaration> fieldsInClass = toTest.findAll(FieldDeclaration.class);
+        String nameOfInterface = interfaceName.getNameAsString();
+        List<String> constructorParams = new ArrayList<>();
+        for (FieldDeclaration currentField : fieldsInClass) {
+            if (currentField.getCommonType().toString().equals(nameOfInterface)) {
+                isInitialized.compareAndSet(true, !currentField.isPublic());
+                if (isInitialized.get()) {
+                    currentField.findAll(InitializerDeclaration.class).forEach(fieldInitializer -> {
+                        isInitialized.set(false);
+                    });
+                    toTest.findAll(ConstructorDeclaration.class).forEach(declaration -> {
+                        Optional<Parameter> parameter = declaration.getParameterByType(
+                            nameOfInterface);
+                        if (parameter.isPresent()) {
+                            constructorParams.add(parameter.get().getNameAsString());
+                        } else {
+                            isInitialized.set(false);
+                        }
+                    });
+                    toTest.findAll(VariableDeclarator.class).forEach(variableDeclarator -> {
+                        if (variableDeclarator.getNameAsString().equals(
+                            currentField.getVariable(0).getNameAsString()) &&
+                            variableDeclarator.getInitializer().isPresent() &&
+                            !constructorParams.contains(
+                                variableDeclarator.getInitializer().get().toString())) {
+                            isInitialized.set(false);
+
+                        }
+                    });
+                }
+            }
+            constructorParams.clear();
+        }
+        if (isInitialized.get()) {
+            result = Feedback.getSuccessfulFeedback();
+        } else {
+            result = Feedback.getNoChildFeedback(
+                "All constructors did not initialize the Component field",
+                new FeedbackTrace(toTest));
+        }
+        return result;
+    }
```

Comment:
I would really like to see these extracted to a seperate class for reuse in other verifications.

Reply:
Sounds reasonable! Should we extract them to PatternUtils?

Reply:
Would rather see a ClassOrInterfaceUtils as PatternUtils are to be used by Pattern, not classes